### PR TITLE
feat(live): add a new resource to manage disable a push stream

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new resource to manage disable a push stream.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new resource to manage disable a push stream.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccDisablePushStream_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccDisablePushStream_basic -timeout 360m -parallel 4
=== RUN   TestAccDisablePushStream_basic
=== PAUSE TestAccDisablePushStream_basic
=== CONT  TestAccDisablePushStream_basic
--- PASS: TestAccDisablePushStream_basic (270.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      270.370s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image
    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image
image

### CODE REVIEW

File: docs/resources/live_disable_push_stream.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+page_title: "HuaweiCloud: huaweicloud_live_disable_push_stream"
+description: |-
+  Manages a disable push stream resource within HuaweiCloud.
+---
+
+# huaweicloud_live_disable_push_stream
+
+Manages a disable push stream resource within HuaweiCloud.
+
+-> **NOTE:** Creating the resource indicates disable a push stream, deleting the resource indicates
```
Comment (jinyangyang222) [inline docs/resources/live_disable_push_stream.md:13]:
```suggestion
-> Creating the resource indicates disable a push stream, deleting the resource indicates
```

File: docs/resources/live_disable_push_stream.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used.
+  Changing this parameter will create a new resource.
+
+* `domain_name` - (Required, String, ForceNew) Specifies the ingest domain name of the disable push stream.
```
Comment (deer-hang) [inline docs/resources/live_disable_push_stream.md]: Please consider the translation carefully, the usage of `of the disable push stream` has grammatical problems.

File: docs/resources/live_disable_push_stream.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID, UUID format.
+
+## Import
+
+The resource can be imported using `domain_name`, `app_name` and `stream_name`, separated by a slash (/), e.g.
```
Comment (deer-hang) [inline docs/resources/live_disable_push_stream.md]:
```suggestion
The resource can be imported using `domain_name`, `app_name` and `stream_name`, separated by slashes (/), e.g.
```

File: huaweicloud/services/acceptance/live/resource_huaweicloud_live_disable_push_stream_test.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	region := acceptance.HW_REGION_NAME
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Live client: %s", err)
+	}
+
+	getRespBody, err := live.GetDisablePushStream(client, state.Primary.Attributes["domain_name"],
+		state.Primary.Attributes["app_name"], state.Primary.Attributes["stream_name"])
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving disable push stream: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/live/resource_huaweicloud_live_disable_push_stream_test.go]:
```suggestion
		return nil, fmt.Errorf("error retrieving disabled push stream: %s", err)
```

File: huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+		KeepResponseBody: true,
+		OkCodes: []int{
+			204,
+		},
+		JSONBody: utils.RemoveNil(buildDisablePushStreamBodyParams(d)),
+	}
+
+	_, err = client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating disable push stream: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go]:
```suggestion
		return diag.Errorf("error creating disabled push stream: %s", err)
```
Please check other similar questions elsewhere.

File: huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func GetDisablePushStream(client *golangsdk.ServiceClient, domainName, appName, streamName string) (interface{}, error) {
+	getHttpUrl := "v1/{project_id}/stream/blocks"
+	getPath := client.Endpoint + getHttpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = fmt.Sprintf("%s?domain=%v&app_name=%v&stream_name=%v", getPath, domainName, appName, streamName)
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go]:
```suggestion
	getPath = fmt.Sprintf("%s?domain=%s&app_name=%s&stream_name=%s", getPath, domainName, appName, streamName)
```

File: huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	deleteHttpUrl := "v1/{project_id}/stream/blocks"
+	deletePath := client.Endpoint + deleteHttpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = fmt.Sprintf("%s?domain=%v", deletePath, domainName)
+	deletePath = fmt.Sprintf("%s&app_name=%v", deletePath, appName)
+	deletePath = fmt.Sprintf("%s&stream_name=%v", deletePath, streamName)
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go]: deletePath = fmt.Sprintf("%s?domain=%s&app_name=%s&stream_name=%s", deletePath, domainName, appName, streamName)

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccDisablePushStream_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccDisablePushStream_basic -timeout 360m -parallel 4
=== RUN   TestAccDisablePushStream_basic
=== PAUSE TestAccDisablePushStream_basic
=== CONT  TestAccDisablePushStream_basic
--- PASS: TestAccDisablePushStream_basic (238.02s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      238.075s
```

**Pull request merged**